src/types/smartSearch.ts

```diff
@@ -18,3 +18,39 @@ export enum TIME_FRAME {
     BETWEEN='between',
     LAST_FEW_DAYS='lastFew',
 }
+
+/**
+ * Filter Configs
+ */
+export type FilterConfig = Record<string, unknown> // Filter Configs are objects
+
+export type DefaultFilterConfig = Record<string, never> // Default filter config is an empty object
+
+export interface MostActiveFilterConfig extends FilterConfig {
+    after?: string;
+    before?: string;
+    size?: number;
+}
+
+export type AnyFilterConfig = DefaultFilterConfig | MostActiveFilterConfig // Add all filter objects here
+
+/** Filters */
+export interface ZetkinSmartSearchFilter<C extends FilterConfig = AnyFilterConfig> {
+    config?: C;
+    op: OPERATION;
+    type: FILTER_TYPE;
+}
+
+export interface SmartSearchFilterWithId<C extends FilterConfig = AnyFilterConfig> extends ZetkinSmartSearchFilter<C> {
+    id: number;
+}
+
+export interface NewSmartSearchFilter {
+    type: FILTER_TYPE;
```

Comment:
I feel like this could still have an `id`.

When you create a new filter, give it a type and an ID and pass it to the edit dialog. The edit dialog doesn't have to care whether it's a new filter being saved for the first time, or an existing one being edited.

Or if it does need to care (e.g. to display different messages) that should be explicit IMO, ideally not derived from the `id`, so that's not a good reason to not always have an ID when passing a filter to the edit dialog.